server/application/application.go

```diff
@@ -1892,7 +1922,7 @@ func (s *Server) RunResourceAction(ctx context.Context, q *application.ResourceA
 		}
 	}
 
-	s.logAppEvent(a, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s on resource %s/%s/%s", q.GetAction(), res.Group, res.Kind, res.Name))
+	s.logAppEvent(a, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s on resource %s/%s/%s", q.GetAction(), q.GetGroup(), q.GetKind(), q.GetResourceName()))
 	s.logResourceEvent(res, ctx, argo.EventReasonResourceActionRan, fmt.Sprintf("ran action %s", q.GetAction()))
```

Comment:
I don't think there is. I think that struct is exclusively intended for resources within an App, rather than the App itself.

Reply:
That makes things easier, as it would make producing a resource audit log for the application nonsensical. 👍🏼

I'll instead return the resource node if there is one, then make this log call conditional on the resource node existing.